compiler/AST/ParamForLoop.cpp

```diff
@@ -61,14 +62,53 @@ BlockStmt* ParamForLoop::buildParamForLoop(VarSymbol* indexVar,
     stride = new SymExpr(new_IntSymbol(1));
   }
 
-  if (call && call->isNamed("chpl_build_bounded_range"))
+  if (call && call->isNamed("#"))
   {
-    low    = call->get(1)->remove();
-    high   = call->get(1)->remove();
+    count       = new CallExpr("chpl_compute_count_param_loop", call->get(2)->remove());
+    call        = toCallExpr(call->get(1));
   }
-  else
+
+  if(call) {
+    if(count == NULL)
+    {
+      // high..low
+      if (call->isNamed("chpl_build_bounded_range"))
+      {
+        low    = call->get(1)->remove();
+        high   = call->get(1)->remove();
+      }
+    }
+    else
+    {
+      // high..low#count
+      if(call->isNamed("chpl_build_bounded_range"))
+      {
+        Expr* temp_low  = call->get(1)->remove();
+        Expr* temp_high = call->get(1)->remove();
+
+        // It is necessary that low is calculated first, because it also applies check for bound size.
+        low  = new CallExpr("chpl_bounded_count_for_param_loop_low", temp_low->copy(), temp_high->copy(), count->copy());
+        high = new CallExpr("chpl_bounded_count_for_param_loop_high", temp_low->copy(), temp_high->copy(), count);
+      }
+      // low..#count
+      else if (count != NULL && call && call->isNamed("chpl_build_low_bounded_range"))
```

Comment:
In this `else if` and the next, I believe we already know that count != NULL and call != NULL, right?

Reply:
Yes, sir. Sorry I missed that.